lib/bolt/transport/winrm/connection.rb

```diff
@@ -408,30 +315,21 @@ def execute(command)
           raise
         end
 
-        # 10 minutes in seconds
-        DEFAULT_EXECUTION_TIMEOUT = 10 * 60
-
-        def execute_process(path = '', arguments = [], stdin = nil,
-                            timeout = DEFAULT_EXECUTION_TIMEOUT)
+        def execute_process(path = '', arguments = [], stdin = nil)
           quoted_args = arguments.map do |arg|
             "'" + arg.gsub("'", "''") + "'"
-          end.join(',')
-
+          end.join(' ')
+
+          exec_cmd =
+            if stdin.nil?
+              "& #{path} #{quoted_args}"
+            else
+              "'#{stdin}' | & #{path} #{quoted_args}"
+            end
           execute(<<-PS)
-$quoted_array = @(
-  #{quoted_args}
-)
-
-$invokeArgs = @{
-  Path = "#{path}"
-  Arguments = $quoted_array -Join ' '
-  Timeout = #{timeout}
-  #{stdin.nil? ? '' : "StdinInput = @'\n" + stdin + "\n'@"}
-}
-
-# winrm gem checks $? prior to using $LASTEXITCODE
-# making it necessary to exit with the desired code to propagate status properly
-exit $(Invoke-Interpreter @invokeArgs)
+#{exec_cmd}
```

Comment:
@Iristyle can you tell me why this is a bad idea? It seems to work for all our current test cases, and addresses the issue.